src/converters/ssm-dynamic.ts

```diff
@@ -0,0 +1,160 @@
+import * as aws from '@pulumi/aws';
+import * as pulumi from '@pulumi/pulumi';
+import { containsEventuals } from '../types';
+
+/**
+ * The regular expression used to match an SSM plaintext dynamic reference.
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references-ssm.html#dynamic-references-ssm-pattern
+ */
+const SSM_PLAINTEXT_DYNAMIC_REGEX = /{{resolve:ssm:([a-zA-Z0-9_.\-/]+(?::\d+)?)}}/;
+
+/**
+ * The regular expression used to match an SSM SecureString dynamic reference.
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references-ssm-secure-strings.html#dynamic-references-ssm-secure-pattern
+ */
+const SSM_SECURE_DYNAMIC_REGEX = /{{resolve:ssm-secure:([a-zA-Z0-9_.\-/]+(?::\d+)?)}}/;
+
+export interface SSMDynamicReference {
+    /**
+     * The name of the parameter you want to reference.
+     * This will also include the version if specified.
+     */
+    parameterName: string;
+}
+
+/**
+ * Parses an SSM plaintext dynamic reference and returns the parameter name.
+ *
+ * @param value - The value which contains the SSM plaintext dynamic reference
+ * @returns The parameter name
+ */
+export function parseSSMDynamicSecureStringReference(value: string): SSMDynamicReference {
+    const match = value.match(SSM_SECURE_DYNAMIC_REGEX);
+    if (!match) {
+        throw new Error(`Failed to parse SSM SecureString dynamic reference: ${value}`);
```

Comment:
Sanity-check: `${value}` will never leak secret material into the error reporting? Probably not.